src/main/java/org/apache/sling/resourceresolver/impl/ResourceResolverImpl.java

```diff
@@ -1117,4 +1142,68 @@ public Resource move(final String srcAbsPath, final String destAbsPath) throws P
         }
         return rsrc;
     }
+
+
+
+    // A very simple tupel implementation which can be used as key in any map
+    public class String3Tupel {
+
+        @Override
+        public int hashCode() {
+            final int prime = 31;
+            int result = 1;
+            result = prime * result + getEnclosingInstance().hashCode();
+            result = prime * result + ((s1 == null) ? 0 : s1.hashCode());
+            result = prime * result + ((s2 == null) ? 0 : s2.hashCode());
+            result = prime * result + ((s3 == null) ? 0 : s3.hashCode());
+            return result;
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            if (this == obj)
+                return true;
+            if (obj == null)
+                return false;
+            if (getClass() != obj.getClass())
+                return false;
+            String3Tupel other = (String3Tupel) obj;
+            if (!getEnclosingInstance().equals(other.getEnclosingInstance()))
+                return false;
+            if (s1 == null) {
+                if (other.s1 != null)
+                    return false;
+            } else if (!s1.equals(other.s1))
+                return false;
+            if (s2 == null) {
+                if (other.s2 != null)
+                    return false;
+            } else if (!s2.equals(other.s2))
+                return false;
+            if (s3 == null) {
+                if (other.s3 != null)
+                    return false;
+            } else if (!s3.equals(other.s3))
+                return false;
+            return true;
+        }
+
+        String s1;
```

Comment:
Any reason to not make these fields private? Also, fields should go before contructors, constructors before instance methods :-)

Reply:
Changed order (not sure what Eclipse did here ...)